gfx.cpp

```diff
@@ -5772,8 +5755,9 @@ class GfxInternal
         for(size_t i = 0; i < kernel.subobjects_.size(); ++i)
             max_export_length = GFX_MAX(max_export_length, strlen(kernel.subobjects_[i].c_str()));
         max_export_length += 1;
-        char *lib_export = (char *)alloca(max_export_length);
-        WCHAR *wexport = (WCHAR *)alloca(max_export_length << 1);
+        char lib_export[1024];
+        WCHAR wexport[1024];
+        GFX_ASSERT(max_export_length < 1024);
```

Comment:
Apparently alloca is not well liked by the compiler, Most other places in gfx use a fixed size array instead of alloca so I changed to that instead

Reply:
Only issue would be running out of memory, but at least there's an assert to catch this.

Although since this is a heavy call anyway (i.e., creating a PSO) we may as well use `gfxMalloc()` here (which you wouldn't want to call on a "hot path" kind of function).

Reply:
Changed to using heap allocated vector (for auto handled free). Modern OSs lazily allocate memory anyway so an out of memory error is never thrown on allocation (unless you exhausted the 64b virtual space) so having to check it doesnt really achieve much. But std::vector with exceptions disabled will just call std::terminate on memory allocation failure so it handles well enough for our needs